### PR TITLE
Grenadier quadratic

### PR DESCRIPTION
**Pull request by Sirkular**
Implemented quadratic motion for grenadier projectiles.
~~Known issue: forcing the grenadiers to move seems to sometimes desync their projectiles. Unsure of the cause.~~ **Fixed.**

### CODE REVIEW

File: Assets/Scripts/Enemy/EnemyFactory.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public EnemyFactory()
     {
         // Define enemyInfo for each type.
         AddTurretToRoster();
-        _enemyInfo.Add(EnemyType.GRENADIER, new EnemyInfo(EnemyMovements.GrenadierMovement, Globals.DirectTargeting, EnemyMovements.GrenadierSetup, new ArtilleryAttackBehaviour(EntityType.ENEMY, 20f), 10f));
+        _enemyInfo.Add(EnemyType.GRENADIER, new EnemyInfo(EnemyMovements.GrenadierMovement, Globals.GrenadierTargeting, EnemyMovements.GrenadierSetup, new ArtilleryAttackBehaviour(EntityType.ENEMY, 20f), 1f));
```
Comment (MaXeraph) [inline Assets/Scripts/Enemy/EnemyFactory.cs:53]: we probably want to revert the attack speed back or make it a bit slower

Comment (Sirkular) [reply Assets/Scripts/Enemy/EnemyFactory.cs]: Period is now 3s.

**Review by zhuandy42 — APPROVED** [review]
lgtm

File: Assets/Scripts/Weapons/BehaviourTypes/ArtilleryAttackBehaviour.cs [hunk 2]
```diff
@@ -17,19 +17,30 @@ public ArtilleryAttackBehaviour(EntityType owner, float damage = 10f, float bull
...
+    
+    public override void initiateAttack(Vector3 position, Vector3 direction)
+    {
+        ArtilleryMono.create(this, position, direction);
+    }
 
     public override void onHit(BulletMono bm, GameObject target)
     {
-        if (target.tag == "Player" || target.GetComponent<BulletMono>() == null) // if not another bullet...
+        if (target.tag != "Enemy" && target.tag != "Detector" && (target.tag == "Player" || target.GetComponent<BulletMono>() == null)) // if not another bullet...
```
Comment (MaXeraph) [inline Assets/Scripts/Weapons/BehaviourTypes/ArtilleryAttackBehaviour.cs:28]: Might want to note this as it might be problematic in the future

**Review by MaXeraph — APPROVED** [review]

**Pull request merged**